packages/base/src/base-element.ts

```diff
@@ -0,0 +1,97 @@
+/**
+@license
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import {LitElement} from '@polymer/lit-element';
+import {TemplateResult} from 'lit-html';
+export {TemplateResult};
+export * from '@polymer/lit-element';
+export {classMap} from 'lit-html/directives/classMap';
+export {observer} from './observer';
+
+/**
+ * Base Adapter class for components.
+ *
+ * For your component extend this component,
+ * put the new element type in the constructor,
+ * and add additional adapters as needed.
+ */
+export class BaseAdapter {
+  constructor(readonly element: BaseElement) {}
+  addClass(className: string) {
+    this.element.mdcRoot.classList.add(className);
+  }
+  removeClass(className: string) {
+    this.element.mdcRoot.classList.remove(className);
+  }
+  hasClass(className: string) {
+    return this.element.mdcRoot.classList.contains(className);
+  }
+}
+
+export abstract class BaseElement extends LitElement {
+
+  /**
+   * Root element for MDC Foundation usage.
+   *
+   * Define in your component with the `@query` decorator
+   */
+  abstract mdcRoot: HTMLElement;
+
+  /**
+   * Return the foundation class for this component
+   */
+  protected abstract get mdcFoundationClass(): any;
```

Comment:
prefer `unknown` over `any`

`any` is contagious. most operations on an `any` produces an `any`, and all operations are allowed, so you can have code far removed from the initial `any` that looks like it's type checked but isn't. this has caused a few bugs in the tools org

in my experience, `unknown` works just as well, occasionally requiring a little more work, but it's way safer by default

Reply:
Working with `unknown` seems like a giant PITA. It might just be easier to describe an interface for the foundations